### DISCUSSION TITLE
How to generate Function Extensions?

### DISCUSSION THREAD

**Discussion by irfansuteja** [tryAGI/LangChain#462] [Q&A]
Hi there, 

In the code, there is WeatherFunctionExtensions that is auto-generated by the generator 'LangChain.Providers.Generators.LangChainToolsGenerator' and cannot be edited. 

If I had another Function Class, how to generate the Extensions?

Thank you.

**Comment by HavenDV** [comment]
You can check these files to understand how to use it. But of course it's better to prepare an article about it for wiki:
https://github.com/tryAGI/LangChain.Providers/blob/717571a98921967bce1f314e2f176bfc31dead2b/src/IntegrationTests/WeatherFunctions.cs
https://github.com/tryAGI/LangChain.Providers/blob/717571a98921967bce1f314e2f176bfc31dead2b/src/IntegrationTests/BookStoreService.cs
https://github.com/tryAGI/LangChain.Providers/blob/682af210c3a0526a56095c7e43a644dd060a560f/src/IntegrationTests/BaseTests.cs

**Reply by irfansuteja** [reply]
Thank you. I'll check the files.
